DisjointSet.java

```diff
@@ -0,0 +1,47 @@
+
+public class DisjointSet {
+	int[] parent;
+	int[] rank;
+
+	public DisjointSet(int numElements) {
+		parent = new int[numElements];
+		for (int i = 0; i < numElements; i++) {
+			parent[i] = i + 1;
```

Comment:
Why do you have the i+1 here and all the x-1s everywhere else? Was the algorithm taken from a language where indexes start at 1.